src/RootPersistencyManager.cxx

```diff
@@ -95,6 +116,68 @@ namespace ldmx {
             runHeader.setFloatParameter( "Biasing Threshold", parameters_.getParameter<double>(     "biasing_threshold" ));
         }
 
+        auto apMass{parameters_.getParameter<double>("APrimeMass")};
+        if ( apMass > 0 ) {
+            runHeader.setFloatParameter(  "A' Mass [MeV]" , apMass );
+            runHeader.setFloatParameter(  "Dark Brem Global Bias" , parameters_.getParameter<double>("darkbrem_globalxsecfactor") );
+            runHeader.setStringParameter( "Dark Brem Vertex Library Path" , parameters_.getParameter<std::string>("darkbrem_madgraphfilepath") );
+            runHeader.setIntParameter(    "Dark Brem Interpretation Method" , parameters_.getParameter<int>("darkbrem_method") );
+        }
+
+        auto generators{parameters_.getParameter<std::vector<Parameters>>("generators")};
+        counter = 0;
+        for ( auto const &gen : generators ) {
+
+            std::string genID = "Gen " + std::to_string(++counter);
+            auto className{gen.getParameter<std::string>("class_name")};
+            runHeader.setStringParameter( genID + " Class" , className );
+
+            if ( className.find("ldmx::ParticleGun") != std::string::npos ) {
```

Comment:
At some point, it might be best to add a callback that injects the run header into the generators.  This will allows us to avoid al of the if-else statements. 

This is fine for now but let's clean it up at some point.